src/Symfony/Component/Translation/Translator.php

```diff
@@ -491,4 +492,19 @@ protected function assertValidLocale($locale)
             throw new \InvalidArgumentException(sprintf('Invalid "%s" locale.', $locale));
         }
     }
+
+    /**
+     * @see WarmableInterface
+     *
+     * @param string $cacheDir only used for interface compatibility, $this->cacheDir is already defined
+     */
+    public function warmUp($cacheDir)
+    {
+        if ($this->locale !== null) {
```

Comment:
i didn't understand way you want to use annotations directory here, but $this->cacheDir from translator hasn't the same value of $cacheDir see https://github.com/symfony/symfony/blob/2.7/src/Symfony/Bundle/FrameworkBundle/Resources/config/translation.xml#L44

```
    public function warmUp($cacheDir)
    {
        if (null !== $this->cacheDir && null !== $this->locale) {
            $this->loadCatalogue($this->locale);
        }
    }
```